autogpt_platform/frontend/src/components/DataTable.tsx

```diff
@@ -17,6 +18,54 @@ type DataTableProps = {
   data: { [key: string]: Array<any> };
 };
 
+const VideoRenderer: React.FC<{ videoUrl: string }> = ({ videoUrl }) => {
+  const getYouTubeVideoId = (url: string) => {
+    const regExp =
+      /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/;
+    const match = url.match(regExp);
+    return match && match[7].length === 11 ? match[7] : null;
+  };
+
+  const videoId = getYouTubeVideoId(videoUrl);
+
+  return (
+    <div className="w-full p-2">
+      {videoId ? (
+        <iframe
+          width="100%"
+          height="315"
+          src={`https://www.youtube.com/embed/${videoId}`}
+          frameBorder="0"
+          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+          allowFullScreen
+        ></iframe>
+      ) : (
+        <video controls width="100%" height="315">
+          <source src={videoUrl} type="video/mp4" />
+          Your browser does not support the video tag.
+        </video>
+      )}
+    </div>
+  );
+};
+
```

Comment:
extract to /ui/ and rename youtube renderer

Reply:
Moved